signature/openpgp/openpgp.go

```diff
@@ -0,0 +1,86 @@
+package openpgp
+
+import (
+	"bytes"
+	"errors"
+	"fmt"
+	"io/ioutil"
+	"strings"
+	"time"
+
+	"github.com/containers/image/types"
+
+	"golang.org/x/crypto/openpgp"
+)
+
+type context struct {
+	keyring openpgp.EntityList
+}
+
+type OpenPGPMechanism struct {
+	ctx *context
+}
+
+func NewOpenPGPSigningMechanism() (types.SigningMechanism, error) {
+	return OpenPGPMechanism{ctx: &context{
+		keyring: openpgp.EntityList{},
+	}}, nil
+}
+
+func (m OpenPGPMechanism) ImportKeysFromBytes(blob []byte) ([]string, error) {
+	keyring, err := openpgp.ReadArmoredKeyRing(bytes.NewReader(blob))
```

Comment:
Considering https://github.com/golang/crypto/blob/master/openpgp/armor/armor.go#L165 , such a trivial check may not be sufficient (PGP implementations have frequently been fairly lenient about leading text before armor, e.g. handling an e-mail message with full headers), though it may affect very few cases in practice.

It is, of course, easy enough, to call both `ReadKeyRing` and `ReadArmoredKeyRing`, in case either succeeds; if both fail, returning a really good error message may be difficult.